app/lib/server/lib/notifyUsersOnMessage.js

```diff
@@ -126,6 +123,9 @@ function notifyUsersOnMessage(message, room) {
 		return message;
 	}
 
+	// Update all the room activity tracker fields
+	Rooms.incMsgCountAndSetLastMessageById(message.rid, 1, message.ts, settings.get('Store_Last_Message') && message);
```

Comment:
this will need to be moved to the top of [updateUsersSubscriptions](https://github.com/RocketChat/Rocket.Chat/pull/14041/files#diff-3dd339746f1a79c1b12aa7d5558bff8cR26) because threads only [call updateUsersSubscriptions](https://github.com/RocketChat/Rocket.Chat/blob/develop/app/threads/server/hooks/aftersavemessage.js#L98).